tests/nvmeof/test_ceph_nvmeof_qos_tests.py

```diff
@@ -0,0 +1,344 @@
+                    **(
+                        {"no-group-append": config.get("no-group-append", True)}
+                        if ceph_cluster.rhcs_version >= "8.0"
+                        else {}
+                    ),
+                },
+            }
+        }
+    )
+    listener_cfg = {
+        "host-name": nvmegwcli.fetch_gateway_hostname(),
+        "traddr": nvmegwcli.node.ip_address,
+        "trsvcid": subsys_config["listener_port"],
+    }
+    nvmegwcli.listener.add(**{"args": {**listener_cfg, **sub_args}})
+    if subsys_config.get("allow_host"):
+        nvmegwcli.host.add(
+            **{"args": {**sub_args, **{"host": repr(subsys_config["allow_host"])}}}
+        )
+
+    if subsys_config.get("hosts"):
+        for host in subsys_config["hosts"]:
+            initiator_node = get_node_by_id(ceph_cluster, host)
+            initiator = Initiator(initiator_node)
+            host_nqn = initiator.nqn()
+            nvmegwcli.host.add(**{"args": {**sub_args, **{"host": host_nqn}}})
+
+    if subsys_config.get("bdevs"):
+        name = generate_unique_id(length=4)
+        with parallel() as p:
+            count = subsys_config["bdevs"].get("count", 1)
+            size = subsys_config["bdevs"].get("size", "1G")
+            # Create image
+            for num in range(count):
+                p.spawn(rbd.create_image, pool, f"{name}-image{num}", size)
+        namespace_args = {**sub_args, **{"rbd-pool": pool}}
+        with parallel() as p:
+            # Create namespace in gateway
+            for num in range(count):
+                ns_args = deepcopy(namespace_args)
+                ns_args.update({"rbd-image": f"{name}-image{num}"})
+                ns_args = {"args": ns_args}
+                p.spawn(nvmegwcli.namespace.add, **ns_args)
+
+    # Set and verify QoS for namespaces
+    if subsys_config.get("bdevs").get("qos"):
+        for qos_args in subsys_config.get("bdevs").get("qos"):
+            prev_key = None
+            for key in [
+                "r-megabytes-per-second",
+                "w-megabytes-per-second",
+                "rw-megabytes-per-second",
+            ]:
+                if key in qos_args:
+                    qos_arg = {
+                        "nsid": qos_args["nsid"],
+                        "subsystem": qos_args["subsystem"],
+                        key: qos_args[key],
+                    }
+
+                    if prev_key:
+                        qos_arg[prev_key] = 0
+
+                    nvmegwcli.namespace.set_qos(
+                        **{
+                            "args": qos_arg,
+                        }
+                    )
+                    verify_qos(qos_arg, nvmegwcli)
+                    prev_key = key
+
+                    run_io_and_validate_qos(
+                        ceph_cluster, config, key, qos_args, nvmegwcli
+                    )
+
+            # Update QoS values
+            qos_arg = {
```

Comment:
suggesting to go with the new definitions fro QOS settings, if the metric values for the NS QOS is set here and also we can remove the NS QOS values from test config, if it is changed later and getting validated.